editor/utils/shortcuts.ts

```diff
@@ -0,0 +1,42 @@
+import { useCallback, useLayoutEffect, useRef, useEffect } from "react";
+
+/**
+ * Custom hook for adding keyboard shortcuts.
+ * 
+ * The shift key must be used in conjunction with the array of keys passed
+ * and the focused element can't be an input or text area
+ * in order to fire the handleKeyPress callback.
+ */
+export const useKeyboardShortcut = (
+  // Array of keys that should trigger the callback when used alongside the shift key
+  shortcutKeys: string[],
+  // Callback that handles what happens when the keys are pressed
+  callback: (event: KeyboardEvent) => void
+) => {
+  const callbackRef = useRef(callback);
+  useLayoutEffect(() => {
+    callbackRef.current = callback;
+  });
+  const handleKeyPress = useCallback(
+    (event: KeyboardEvent) => {
+      // Type guard to access tagName property, see https://iifx.dev/en/articles/156175355
+      if (event.target instanceof Element) {
+        // Don't trigger callback if focused element is an input or textarea
+        if (event.target.tagName === "INPUT" || event.target.tagName === "TEXTAREA") {
```

Comment:
this should cover all of our text input elements on the crash details page per my testing

Reply:
nice! big TIL for me here. this is great.

it might be good to add `SELECT` elements to this test as well. i know that Xavier uses the keyboard to navigate dropdown lists by pressing a letter to find the first option that starts with that letter. the `shift` key probably makes it a real edge case, but might as well be safe.

Reply:
great call, added that!